### PR TITLE
[Meta] Add Mjolnir Cog

### PR DESCRIPTION
**Pull request by Just-Jojo**
### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature
- [x] Documentation

### Description of the changes

Add Mjolnir :D

### CODE REVIEW

File: mjolnir/info.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "name": "mjolnir",
+    "short": "Try to lift Thor's mighty hammer!",
+    "description": "Attempt to lift Thor's hammer to become the next wielder of Mjolnir",
+    "end_user_data_statement": "This cog stores a number based on times lifted via user ids",
+    "install_msg": "Thanks for installing Mjolnir! Have fun!",
+    "author": [
+        "Jojo#7791"
+    ],
```
Comment (Kreusada) [inline mjolnir/info.json:9]:
```suggestion
    "author": ["Kreusada", "Jojo"],
```

File: mjolnir/info.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "install_msg": "Thanks for installing Mjolnir! Have fun!",
+    "author": [
+        "Jojo#7791"
+    ],
+    "required_cogs": {},
+    "requirements": [],
+    "tags": [
+        "fun",
+        "minigame"
+    ],
```
Comment (Kreusada) [inline mjolnir/info.json:15]:
```suggestion
    "tags": ["Fun", "Marvel"],
```

File: mjolnir/menus.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
```
Comment (Kreusada) [inline mjolnir/menus.py:23]:
```suggestion
# 99% of this code belongs to Jojo, which was authored under the MIT license.
# This cog was transferred to Kreusada (12/04/2021)
# https://github.com/Just-Jojo/JojoCogs
```

I'd rather have a secluded note than a license, but this is optional to you.

File: mjolnir/mjolnir.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
```
Comment (Kreusada) [inline mjolnir/mjolnir.py:23]:
```suggestion
# This code originally belonged to Jojo, which was authored under the MIT license.
# This cog was transferred to Kreusada (12/04/2021)
# https://github.com/Just-Jojo/JojoCogs
```
I'd rather have secluded notes than a license, but this is optional to you.

File: mjolnir/mjolnir.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    "The hammer is strong, but so are you. Keep at it!",
+    "Mjolnir budges a bit, but remains steadfast, as should you",
+    "You've got this! I believe in you!",
+    "Don't think it even moved... why don't you try again?",
+)
+
+class Mjolnir(commands.Cog):
+    """Attempt to lift Thor's hammer!"""
+
+    __version__ = "0.1.1"
```
Comment (Kreusada) [inline mjolnir/mjolnir.py:49]:
```suggestion
    __version__ = "0.1.1"
    __author__ = ["Jojo", "Kreusada"]
```

File: mjolnir/mjolnir.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.config = Config.get_conf(self, 1242351245243535476356, True)
+        self.config.register_user(lifted=0)
+
```
Comment (Kreusada) [inline mjolnir/mjolnir.py:55]:
```suggestion

    def format_help_for_context(self, ctx: commands.Context) -> str:
        context = super().format_help_for_context(ctx)
        authors = ", ".join(self.__author__)
        return f"{context}\n\nAuthor: {authors}\nVersion: {self.__version__}"
        
```

File: mjolnir/mjolnir.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    async def red_delete_data_for_user(self, *, requester, user_id):
+        await self.config.user_from_id(user_id).clear()
+
+    @commands.command()
+    async def lifted(self, ctx):
+        """Shows how many times you've lifted the hammer."""
+        lifted = await self.config.user(ctx.author).lifted()
+        plural = "s" if lifted != 1 else ""
+        await ctx.send(f"You have lifted Mjolnir {lifted} time{plural}")
```
Comment (Kreusada) [inline mjolnir/mjolnir.py:64]:
```suggestion
        await ctx.send(f"You have lifted Mjolnir {lifted} time{plural}.")

```

It's my OCD, sorry :P

File: mjolnir/mjolnir.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+    @commands.cooldown(1, 60.0, commands.BucketType.user)
+    @commands.command()
+    async def trylift(self, ctx):
+        """Try and lift Thor's hammer!"""
+        lifted = random.randint(0, 100)
+        if lifted >= 95:
+            await ctx.send(
+                "The sky opens up and a bolt of lightning strikes the ground\nYou are worthy. Hail, son of Odin"
```
Comment (Kreusada) [inline mjolnir/mjolnir.py:73]:
```suggestion
                "The sky opens up and a bolt of lightning strikes the ground\nYou are worthy. Hail, son of Odin."

```

**Review by Kreusada — CHANGES_REQUESTED** [review]
A few changes which I would like to see to unify with the style of my other cogs. They are very small changes which can be committed straight from this review.

I'd also like you to add yourself to the CODEOWNERS file in the root of this repository.

```CODEOWNERS
/mjolnir/* @Just-Jojo 
```

I'd also appreciate adding the labelers. (Yes, I forgot minifier when I had released it so please include that too. You can find this file in `.github/labeler.yml`

```yaml
"Cog: Minifier":
  - minifier/*
  - docs/cog_minifier
  - docs/changelog_minifier
"Cog: Mjolnir":
  - mjolnir/*
  - docs/cog_mjolnir
  - docs/changelog_mjolnir
```

I can take care of the documentation of this cog before it is published onto this repository.

**Comment by Just-Jojo** [comment]
Made the requested changes!

**Comment by Kreusada** [comment]
Perfect, just gotta write the docs and then we're to go. 🎉

**Review by Kreusada — APPROVED** [review]

**Comment by Kreusada** [comment]
All set now. I've got the documentation in the `docs/` directory which will be merged along with this PR. This PR also won't be merged until I have finished and approved #82 and #83.

**Review by Kreusada — APPROVED** [review]
Thanks for contributing!

**Pull request merged**